core/src/main/java/org/mule/runtime/core/internal/context/DefaultMuleContext.java

```diff
@@ -1537,6 +1539,16 @@ private static void configureForeachRouterRejectsMapExpressionFeatureFlag() {
     featureFlaggingRegistry.registerFeatureFlag(FOREACH_ROUTER_REJECTS_MAP_EXPRESSIONS, minMuleVersion("4.5.0"));
   }
 
+  /**
+   * Configures the {@link MuleRuntimeFeature#USE_TRANSACTION_SINK_INDEX} feature flag.
+   *
+   * @since 4.0.0
+   */
+  private static void configureUseTransactionSinkIndex() {
+    FeatureFlaggingRegistry featureFlaggingRegistry = FeatureFlaggingRegistry.getInstance();
+    featureFlaggingRegistry.registerFeatureFlag(USE_TRANSACTION_SINK_INDEX, minMuleVersion("4.0.0"));
```

Comment:
change this to alsways return -> true, featureContext -> true instead of minMuleVersion("4.0.0) (this is the best way to do it, my mistake)